src/audio/module_adapter/module_adapter.c

```diff
@@ -844,6 +911,25 @@ int module_adapter_trigger(struct comp_dev *dev, int cmd)
 		return PPL_STATUS_PATH_STOP;
 	}
 
+	if (mod->num_input_buffers > 1) {
+		bool sources_active;
+
+		sources_active = module_source_status_count(dev, COMP_STATE_ACTIVE) ||
+				 module_source_status_count(dev, COMP_STATE_PAUSED);
```

Comment:
I'd rename the variable as `are_sources_active` as `sources_active` could be interpreted as a counter, which makes the `||` operator really weird here.